### PR TITLE
Feature testing

### PR DESCRIPTION
**Pull request by aldemeery**
### Added

- Test the engine class.
- Test the player class.
- Test the game design.
- Game design refactoring.

### CODE REVIEW

**Comment by aldemeery** [comment]
Changed the base branch

File: spec/engine_spec.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      expect(engine.winning?(board, 3)).to eql(true)
+    end
+
+    it 'returns false when there is no winner' do
+      board = Array.new(9)
+      board[0] = firstPlayer
+
+      expect(engine.winning?(board, 1)).to eql(false)
+    end
+  end
```
Comment (majovanilla) [inline spec/engine_spec.rb:27]: This method is calling some private methods you have on the Engine class.  Although there is no need to test private methods, you do need to test your winning method for all that it is checking. So you need to:
- check for a winner in a row
- check for a winner in a column
- check for a winner in a diagonal

Comment (aldemeery) [reply spec/engine_spec.rb]: Done

**Comment by aldemeery** [comment]
Hi @majovanilla 
Thank you for your recommendation about refactoring the game design and moving the logic from `main.rb` into `game.rb`
I have taken that into consideration and made some changes to my code.

**Comment by aldemeery** [comment]
Isses fixed.

**Pull request merged**